### PR TITLE
add editorial for cf 1368D

### PR DESCRIPTION
**Pull request by 876pol**
_If any of the below don't apply to this Pull Request, mark the checkbox as done._

- [ ] I have tested my code.
- [ ] I have added my solution according to the steps [here](https://usaco.guide/general/adding-solution#steps).
- [ ] I have followed the code conventions mentioned [here](https://usaco.guide/general/adding-solution/#code-conventions).
  - I understand that if it is clear that I have not attempted to follow these conventions, my PR will be closed.
  - If changes are requested, I will re-request a review after addressing them.

### CODE REVIEW

File: solutions/silver/cf-1368D.mdx [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+[Official Analysis](https://codeforces.com/blog/entry/79027)
+
+# Explanation
+
+**Observation 1**: Notice that $x \ AND \ y + x \ OR \ y = x + y$, so the total sum of all the elements never changes.
+
+**Observation 2**: Notice that during each operation, the number and position of the set bits don't change. For instance, let $x = 110_2$, $y = 101_2$. $x \ AND \ y = 100_2$, and $x \ OR \ y = 111_2$. Before and after the operation, there are two $1$ bits that are set, one $2$ bit that is set, and one $4$ bit that is set.
+
+**Observation 3**: Notice when performing an operation, $\sum_{i=1}^{n}{a_i^2}$ increases when $|x - y|$ increases. Thus, we want to keep performing operations such that $|x - y|$ increases. In fact, because $x \ AND \ y \leq \min(x, y)$ and $x \ OR \ y \geq \max(x, y)$, performing an operation will never decrease $|x - y|$.
+
```
Comment (ryanchou-dev) [inline solutions/silver/cf-1368D.mdx:17]: it'd be great if you could structure these as hints! (see https://github.com/cpinitiative/usaco-guide/pull/3430)

File: solutions/silver/cf-1368D.mdx [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+	// Count the number of set bits at each position.
+	vector<int> num_bits(20);
+	for (int i = 0; i < 20; i++) {
+		for (int j = 0; j < n; j++) {
+			if (a[j] & (1ll << i)) { num_bits[i]++; }
+		}
+	}
+
+	// Create the optimal array and calculate the sum of the squares of the new
+	// array.
```
Comment (ryanchou-dev) [inline solutions/silver/cf-1368D.mdx:50]: make multiline + even lines out?

**Review by ryanchou-dev — COMMENTED** [review]
awesome :O

File: solutions/silver/cf-1368D.mdx (hunk 2 above)
Comment (SansPapyrus683) [reply solutions/silver/cf-1368D.mdx:50]: ^

File: solutions/silver/cf-1368D.mdx [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+
+</CPPSection>
+
+<JavaSection>
+
+```java
+import java.io.*;
+import java.util.*;
+
+class AndOrSquareSum {
```
Comment (SansPapyrus683) [inline solutions/silver/cf-1368D.mdx:74]:
```suggestion
public class AndOrSquareSum {
```

File: solutions/silver/cf-1368D.mdx [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+	public static void main(String[] args) {
+		Scanner sc = new Scanner(System.in);
```
Comment (SansPapyrus683) [inline solutions/silver/cf-1368D.mdx:76]: use bufferedreader/kattio for consistency

File: solutions/silver/cf-1368D.mdx [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+for i in range(20):
+	for j in range(n):
+		if a[j] & (1 << i):
+			num_bits[i] += 1
+
+# Create the optimal array and calculate the sum of the squares of the new array.
+ans = 0
+for i in range(n):
+	curr = 0
+	for j in range(20):
```
Comment (SansPapyrus683) [inline solutions/silver/cf-1368D.mdx:128]: put in constant

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]
poggies

File: solutions/silver/cf-1368D.mdx [hunk 6]
```diff
@@ -0,0 +1,139 @@
...
+author: Paul Chen
+---
+
+[Official Analysis](https://codeforces.com/blog/entry/79027)
+
+# Explanation
+
+**Observation 1**: Notice that $x \ AND \ y + x \ OR \ y = x + y$, so the total sum of all the elements never changes.
+
+**Observation 2**: Notice that during each operation, the number and position of the set bits don't change. For instance, let $x = 110_2$, $y = 101_2$. $x \ AND \ y = 100_2$, and $x \ OR \ y = 111_2$. Before and after the operation, there are two $1$ bits that are set, one $2$ bit that is set, and one $4$ bit that is set.
```
Comment (jessechoe10) [inline solutions/silver/cf-1368D.mdx:14]: > the number and position of the set bits don't change
and
> Before and after the operation, there are two $1$ bits that are set, one $2$ bit that is set, and one $4$ bit that is set.
doesn't really make sense

File: solutions/silver/cf-1368D.mdx [hunk 7]
```diff
@@ -0,0 +1,139 @@
...
+source: CF
+title: AND, OR and square sum
+author: Paul Chen
+---
+
+[Official Analysis](https://codeforces.com/blog/entry/79027)
+
+# Explanation
+
+**Observation 1**: Notice that $x \ AND \ y + x \ OR \ y = x + y$, so the total sum of all the elements never changes.
```
Comment (jessechoe10) [inline solutions/silver/cf-1368D.mdx:12]: It'd be great if you could prove this claim

File: solutions/silver/cf-1368D.mdx [hunk 8]
```diff
@@ -0,0 +1,139 @@
...
+
+Therefore, the optimal configuration occurs when it is impossible to change $|x - y|$ with more operations. This happens when, for any two values $a_i$ and $a_j$, $a_i \ AND \ a_j = \min(a_i, a_j)$, and $a_i \ OR \ a_j = \max(a_i, a_j)$. In other words, when we sort the optimal configuration in ascending order, $a_i \ AND \ a_{i + 1} = a_i$.
+
+The optimal configuration can be built by handling each bit separately. This approach takes into account Observation 2, which states that no operations will alter the bits. For instance, consider the array, `[1, 5, 4, 7, 3]`. In binary form, it is represented as `[001, 101, 100, 111, 011]`. To structure this array optimally, move all the set bits to one end. This results in the reconfigured array: `[000, 001, 101, 111, 111]`. In decimal form, this new array reads `[0, 1, 5, 7, 7]`. Finally, find the sum of squares of the elements in this new array. The resulting total provides the desired solution.
```
Comment (jessechoe10) [inline solutions/silver/cf-1368D.mdx:20]: > To structure this array optimally, move all the set bits to one end.

What does set bits mean and also explain exactly how the array is reconfigured given you provided an example

Comment (jessechoe10) [reply solutions/silver/cf-1368D.mdx:20]:
```suggestion
The optimal configuration can be built by handling each bit separately. This approach takes into account Observation 2, which states that no operations will alter the bits. For example, consider the array, `[1, 5, 4, 7, 3]`. In binary form, it is represented as `[001, 101, 100, 111, 011]`. To structure this array optimally, move all the set bits to one end. This results in the reconfigured array: `[000, 001, 101, 111, 111]`. In decimal form, this new array reads `[0, 1, 5, 7, 7]`. Finally, find the sum of squares of the elements in this new array. The resulting total provides the desired solution.
```

File: solutions/silver/cf-1368D.mdx [hunk 9]
```diff
@@ -0,0 +1,139 @@
...
+#include <bits/stdc++.h>
+using namespace std;
+
+int main() {
+	int n;
+	cin >> n;
+	vector<int> a(n);
+	for (int i = 0; i < n; i++) { cin >> a[i]; }
+
+	// Count the number of set bits at each position.
```
Comment (jessechoe10) [inline solutions/silver/cf-1368D.mdx:41]: Reword this

File: solutions/silver/cf-1368D.mdx [hunk 10]
```diff
@@ -0,0 +1,139 @@
...
+		// Count the number of set bits at each position.
+		int[] num_bits = new int[20];
+		for (int i = 0; i < 20; i++) {
+			for (int j = 0; j < n; j++) {
+				if ((a[j] & (1L << i)) != 0) { num_bits[i]++; }
+			}
+		}
+
+		// Create the optimal array and calculate the sum of the squares of the
+		// new array.
```
Comment (jessechoe10) [inline solutions/silver/cf-1368D.mdx:91]: multiline + make each line even

**Comment by 876pol** [comment]
i made a few changes to the explanation - lmk if some parts are still unclear

File: solutions/silver/cf-1368D.mdx [hunk 11]
```diff
@@ -0,0 +1,199 @@
...
+
+# Count the number of active bits at each position.
+num_bits = [0] * MAX_BIT
+for i in range(MAX_BIT):
+	for j in range(n):
+		if a[j] & (1 << i):
+			num_bits[i] += 1
+
+# Create the optimal array and calculate
+# the sum of the squares of the new array.
```
Comment (ryanchou-dev) [inline solutions/silver/cf-1368D.mdx:184]: mutiline comment this

Comment (SansPapyrus683) [reply solutions/silver/cf-1368D.mdx:184]: if it's two lines it's fine

File: solutions/silver/cf-1368D.mdx [hunk 12]
```diff
@@ -0,0 +1,199 @@
...
+
+Summing each of these bit positions gives us $x \ AND \ y + x \ OR \ y = x + y$.
+
+</Spoiler>
+
+</Spoiler>
+
+<Spoiler title="Hint 2">
+
+Notice that before and after each operation, the amount of active bits at each position don't change.
```
Comment (jessechoe10) [inline solutions/silver/cf-1368D.mdx:30]:
```suggestion
Notice that before and after each operation, the number of active bits at each position don't change.
```

File: solutions/silver/cf-1368D.mdx [hunk 13]
```diff
@@ -0,0 +1,199 @@
...
+
+For example, let $x = 110_2$, $y = 101_2$. $x \ AND \ y = 100_2$, and $x \ OR \ y = 111_2$.
+
+- The rightmost bit of $y$ is active, as well as the rightmost bit of $x \ OR \ y$.
+
+- The second rightmost bit of $x$ is active, as well as the second rightmost bit of $x \ OR \ y$.
```
Comment (jessechoe10) [inline solutions/silver/cf-1368D.mdx:36]: same here

File: solutions/silver/cf-1368D.mdx [hunk 14]
```diff
@@ -0,0 +1,199 @@
...
+
+- The third rightmost bits of $x$ and $y$ are active, as well as the third rightmost bits of $x \ AND \ y$ and $x \ OR \ y$.
```
Comment (jessechoe10) [inline solutions/silver/cf-1368D.mdx:38]: and here

File: solutions/silver/cf-1368D.mdx [hunk 15]
```diff
@@ -0,0 +1,199 @@
...
+
+Notice that $x \ AND \ y + x \ OR \ y = x + y$, so the total sum of all the elements never changes.
+
+<Spoiler title="Why is this true?">
+
+Let's analyze how much each bit position contributes to the sum. For every $i$, let $A$ be the $i$-th rightmost bit of $x$ and $B$ be the $i$-th rightmost bit of $y$ ($A, B \in \{0, 1\}$).
+
+By trying out all possible cases for $a$ and $b$, we get that $A \ AND \ B + A \ OR \ B = A + B$.
+
+Summing each of these bit positions gives us $x \ AND \ y + x \ OR \ y = x + y$.
```
[the remark this answers is not in the document]
Comment (876pol) [reply solutions/silver/cf-1368D.mdx]: i wasn't too sure how to draw it out as a venn diagram, but i added a truth table to make it more clear

**Comment by SansPapyrus683** [comment]
image

💀

**Comment by 876pol** [comment]
😳 should work now

**Review by jessechoe10 — APPROVED** [review]
I'll wait for Sans to approve before merging.

**Review by ryanchou-dev — COMMENTED** [review]
would be cool if you could use `&` and `|` to keep it consistent!

other than that, looks good. we rly don't need 3 content managers to approve, so ill merge afterwards :DD

**Pull request merged**